Don't crash dropdown menu when selectedIndex is less than 0

jeroencoumans opened the pull request:
Fixes 'Uncaught TypeError: Cannot read property 'text' of undefined' when passing an index of -1.

oliviertassinari commented on src/js/drop-down-menu.jsx:
```diff
@@ -48,12 +48,14 @@ var DropDownMenu = React.createClass({
       'mui-open': this.state.open
     });
 
+    var selectedIndex = this.state.selectedIndex > -1 ? this.state.selectedIndex : 0;
```
I would rather right

var selectedIndex = this.state.selectedIndex;
selectedIndex = (selectedIndex > -1)  ? selectedIndex : 0;

for a better optimisation with uglifyjs

mmrtnz commented:
Can you include a console warning when users pass a negative index? Negative indices are typically due to poorly written code or bugs. A warning will notify users of the potential error. I would also follow @oliviertassinari's suggestion. 

Please follow [these examples](https://github.com/callemall/material-ui/pull/349/files) when writing the warning.

This PR will be good to go after those two changes are made.

jeroencoumans commented:
I've changed it according to your comments. I also moved the check to componentWillReceiveProps, which seems more appropriate than doing this in render.

oliviertassinari commented:
Be aware that componentWillReceiveProps is not called for the initial render.

jeroencoumans commented:
I've made a separate method that's called on mounting and when receiving props. Merged and pulled and rebased against master... let me know if this is fine :)

mmrtnz commented:
Thanks @jeroencoumans!
